parquet/api.py

```diff
@@ -0,0 +1,280 @@
+        self.schema = fmd.schema
+        self.row_groups = fmd.row_groups or []
+        self.key_value_metadata = fmd.key_value_metadata
+        self.created_by = fmd.created_by
+        self.group_files = {}
+        for i, rg in enumerate(self.row_groups):
+            for chunk in rg.columns:
+                self.group_files.setdefault(i, set()).add(chunk.file_path)
+
+    @property
+    def columns(self):
+        return [f.name for f in self.schema if f.num_children is None]
+
+    def _read_partitions(self):
+        cats = {}
+        for rg in self.row_groups:
+            for col in rg.columns:
+                partitions = re.findall("([a-zA-Z_]+)=([^/]+)/",
+                                        col.file_path or "")
+                for key, val in partitions:
+                    cats.setdefault(key, set()).add(val)
+        self.cats = {key: list(v) for key, v in cats.items()}
+
+    def to_dask(self, columns=None, usecats=None, **kwargs):
+        import dask.dataframe as dd
+        cols = columns or (self.columns + list(self.cats))
+        tot = [self.read_row_group_delayed(rg, cols, usecats, **kwargs)
+               for rg in self.row_groups]
+        dtypes = {k: v for k, v in self.dtypes.items() if k in cols}
+
+        # TODO: if categories vary from one rg to next, need to cope
+        return dd.from_delayed(tot, metadata=dtypes, divisions=self.divisions)
+
+    def read_row_group_delayed(self, rg, cols, usecats, filters={}):
+        from dask import delayed
+        return delayed(self.read_row_group)(rg, cols, usecats, filters=filters)
+
+    def read_row_group(self, rg, cols, usecats, filters={}):
+        """Filter syntax: [(column, op, val), ...],
+        where op is [==, >, >=, <, <=, !=, in, not in]
+        """
+        out = {}
+        fname = self.fname
+        helper = schema.SchemaHelper(self.schema)
+        if filters and filter_out_stats(rg, filters, helper):
+            return pd.DataFrame()
+        if filters and self.cats and filter_out_cats(rg, filters):
+            return pd.DataFrame()
+
+        for col in rg.columns:
+            name = ".".join(col.meta_data.path_in_schema)
+            se = helper.schema_element(name)
+            if name not in cols:
+                continue
+
+
+            if col.file_path is None:
+                # continue reading from the same base file
+                infile = self.f
+            else:
+                # relative file
+                ofname = os.path.join(os.path.dirname(self.fname),
+                                      col.file_path)
+                if ofname != fname:
+                    # open relative file, if not the current one
+                    infile = self.open(ofname)
+                    fname = ofname
+
+            use = name in usecats if usecats is not None else False
+            s = core.read_col(col, helper, infile, use_cat=use)
+            out[name] = s
+        out = pd.DataFrame(out)
```

Comment:
Does this maintain the same column ordering?  Perhaps also explicitly provide the columns list?